index.js

```diff
@@ -81,7 +83,14 @@ module.exports = function (opts) {
 					deleteRequireCache(modId);
 				});
 			}
-			jasmine.addReporter(new SilentReporter(cb, errorOnFail));
+			jasmine.addReporter(new SilentReporter(function (error) {
+				if (error) {
+					cb(error);
+				} else {
+					self.emit('end');
+					cb();
```

Comment:
I got it! I've finally got it! Took me long enough, but I understand what's happening. I'm sick today (which might have contributed to my epiphany), but I'll prepare something for tomorrow.